parser/parser.go

```diff
@@ -1037,6 +1037,15 @@ func (p *Parser) parseStmt() stmt.Stmt {
 		}
 		p.expectSemi()
 		return s
+	case token.Defer:
+		pos := p.pos()
+		p.next()
+		s := &stmt.Defer{
+			Position: pos,
+			Expr:     p.parseExpr(),
+		}
```

Comment:
There are strict requirements on the kind of expression that can be passed to defer. You could enforce them here by checking to see if s.Expr is of type (*expr.Call).

Reply:
done.